lib/dfe/analytics/entity_table_check_job.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module DfE
+  module Analytics
+    # Reschedules to run every 24hours
+    class EntityTableCheckJob < AnalyticsJob
+      WAIT_TIME = 24.hours
+
+      def perform
+        DfE::Analytics.entities_for_analytics.each do |entity_name|
+          DfE::Analytics.models_for_entity(entity_name).each do |model|
+            entity_table_check_event =
+              DfE::Analytics::Event.new
+                                   .with_type('entity_table_check')
+                                   .with_entity_table_name(model.table_name)
+                                   .with_data(entity_table_check_data(model))
+                                   .as_json
+
+                         
+
+            DfE::Analytics::SendEvents.perform_later([entity_table_check_event])
+
+            Rails.logger.info("Processing data for #{model.table_name} with row count #{model.count}") 
+          end  
+        end
+        reschedule_job
+      end
+
+      def entity_table_check_data(model)
+        {
+          number_of_rows: model.count,
+          checksum: checksum(model)
+        }
+      end
+
+      def reschedule_job
+        self.class.set(wait: WAIT_TIME).perform_later
```

Comment:
I think this is good for now. We can start off with a checksum running at midnight and maybe add config later if required. It might be that night checksum clashes with other scheduled tasks on some of the services, in which case a custom time might be required, but that's a real requirement yet.

One thing we may want to do is add config to enable / disable the nightly checksum, just in case it causes any issues. Maybe enable by default ?

Reply:
Now added 👍